test/utils.js

```diff
@@ -0,0 +1,70 @@
+import { watchFile, readdirSync, statSync } from 'fs';
+import path from 'path';
+
+/**
+ * Ignored directories.
+ */
+
+var ignore = [ 'node_modules', '.git' ];
+
+/**
+ * Taken from: https://github.com/mochajs/mocha/blob/8a37e016e5629b46709891740c74d5f0b69929e0/lib/utils.js#L168
+ *
+ * Watch the given `files` for changes
+ * and invoke `fn(file)` on modification.
+ *
+ * @param {Array} files the array of files
+ * @param {Function} fn the function to call on modification
+ */
+exports.watch = function( files, fn ) {
+	var options = { interval: 100 };
+	files.forEach( function( file ) {
+		watchFile( file, options, ( curr, prev ) => {
+			if ( prev.mtime < curr.mtime ) {
+				fn( file );
+			}
+		} );
+	} );
+};
+
+/**
+ * Taken from: https://github.com/mochajs/mocha/blob/8a37e016e5629b46709891740c74d5f0b69929e0/lib/utils.js
+ *
+ * Ignored files.
+ *
+ * @param {string} fp file path
+ * @return {boolean} true if should ignore
+ */
+function ignored( fp ) {
+	return ! ~ignore.indexOf( fp );
```

Comment:
>We become tied to Mocha's implementation (undocumented functionality that they have no promise to support/keep). Let me know what you think!

This is a fair point, and maybe we should consider exploring other utilities that offer this functionality as their stated purpose (e.g. [gaze](npmjs.com/package/gaze)). That said, bringing in a new dependency has its own trade-offs in terms of maintainability. Since there's only a handful of functions here, it might be reasonable to copy them from Mocha, though as stated above, there's several styling concerns to address in doing so.